view/frontend/templates/init_validation_modal.phtml

```diff
@@ -13,31 +13,29 @@
  * @license    http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
  */
 ?>
-<script type="text/x-magento-init">
+<?= $secureRenderer->renderTag('script', ['type' => 'text/x-magento-init'], '
     {
         "*": {
             "addressValidationModal": {
-                "validationEnabled": <?= (int)$this->isValidationEnabled() ?>,
-                "hasChoice": <?= $this->getChoice() ?>,
-                "instructions": <?= $this->getInstructions() ?>,
-                "errorInstructions": <?= $this->getErrorInstructions() ?>,
-                "countriesEnabled": "<?= $this->getCountriesEnabled() ?>"
+                "validationEnabled": ' . (int)$this->isValidationEnabled() . ',
+                "hasChoice": "' . $this->getChoice() . '",
+                "instructions": "' . $this->getInstructions() . '",
+                "errorInstructions": "' . $this->getErrorInstructions() . '",
+                "countriesEnabled": "' . $this->getCountriesEnabled() . '"
             }
         }
     }
-</script>
-<script>
+', false); ?>
+
+<?= $secureRenderer->renderTag('script', ['type' => 'text/javascript'], "
     require([
         'mage/url'
     ], function(url) {
-        <?php
         // Set baseUrl so that the urlBuilder JS module has access to it
         // Calling getUrl instead of getBaseUrl as the latter has been overridden in
         // \ClassyLlama\AvaTax\Block\CustomerAddress to return a different url
-        ?>
-        return url.setBaseUrl('<?php /* @escapeNotVerified */ echo $block->getUrl();?>');
-    })
-</script>
-<script type="text/javascript">
-    window.avaTaxStoreCode = "<?= $this->getStoreCode() ?>";
-</script>
+        return url.setBaseUrl('" . $block->getUrl() . "');
+    });
+", false); ?>
+
+<?= $secureRenderer->renderTag('script', ['type' => 'text/javascript'], "\nwindow.avaTaxStoreCode = " . $this->getStoreCode() . ";\n", false); ?>
```

Comment:
@freak3dot - Thanks for this. BTW, in this line, the rendered version doesn't wrap the store code in double-quotes. Implemented as-is, it outputs the following:

`window.avaTaxStoreCode = whatever_store_view;`

That gives a JS error. It needs to be rewritten like this (note the two new backslashed double-quotes):

`<?= $secureRenderer->renderTag('script', ['type' => 'text/javascript'], "\nwindow.avaTaxStoreCode = \"" . $this->getStoreCode() . "\";\n", false); ?>`

This will also need to be updated in 2 other places in this PR. So three places in total.

Reply:
I think this pull request is irrelevant now. Avalara seems to have fixed CSP issues in version 2.8.2 of their extension. I didn't realize that the newer versions of the extension were not on GitHub until after creating the pull request.

Referenced: https://commercemarketplace.adobe.com/avalara-avatax-magento.html#product.info.details.release_notes

Reply:
@freak3dot - I didn't either until just now. Thanks for the info!